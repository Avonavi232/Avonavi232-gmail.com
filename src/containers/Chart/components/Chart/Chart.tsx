import * as React from 'react';

import { ChartLegend } from '../ChartLegend';

import * as styles from './styles.module.css';

export interface IChartProps {
  chartCanvasRef: React.RefObject<HTMLCanvasElement>;
}

export const Chart: React.FC<IChartProps> = ({ chartCanvasRef }) => {
  return (
    <>
      <div className={styles['legend']}>
        <ChartLegend />
      </div>
      <div className={styles['chart']} style={{ height: 186 }}>
        <canvas ref={chartCanvasRef} />
      </div>
    </>
  );
};

Chart.displayName = 'Chart';
