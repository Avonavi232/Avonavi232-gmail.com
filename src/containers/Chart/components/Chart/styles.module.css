.legend {
  margin-bottom: 10px;
}

.chart {
  display: block;
  position: relative;
}
