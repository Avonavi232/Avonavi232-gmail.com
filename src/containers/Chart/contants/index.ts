export const dayInMs = 86400000;

export const chartHeight = 186;

export const chartLegend = [
  { key: 'offerPrice', name: 'В объявлении', color: '#FF1F34' },
  { key: 'estimatedPrice', name: 'Рыночная цена', color: '#0468FF' },
];

export const months = ['янв', 'фев', 'мар', 'апр', 'май', 'июн', 'июл', 'авг', 'сен', 'окт', 'ноя', 'дек'];
