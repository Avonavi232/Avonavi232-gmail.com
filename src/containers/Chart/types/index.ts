export interface IChartData {
  labels: Array<string | number>;
  estimatedPriceData: Array<number | null>;
  offerPriceData: Array<number | null>;
  minChartPrice?: number;
  maxChartPrice?: number;
}

export interface IChartContainerProps {
  chartOptions: IChartData;
}

export interface IMarketPriceProps {
  price: string;
}

export interface IChartScale {
  ticks: Array<{ label: string | number; value: string | number }>;
  // tslint:disable-next-line:no-any
  [key: string]: any;
}

export interface ITick {
  label?: string | number;
}
