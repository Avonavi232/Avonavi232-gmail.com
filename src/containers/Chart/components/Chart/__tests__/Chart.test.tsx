import * as React from 'react';
import { shallow } from 'enzyme';

import { Chart } from '../Chart';

describe('Chart', () => {
  it('должен рендериться без ошибок', () => {
    const wrapper = shallow(<Chart chartCanvasRef={{ current: null }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
