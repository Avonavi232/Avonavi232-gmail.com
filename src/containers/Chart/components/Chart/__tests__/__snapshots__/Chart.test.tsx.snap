// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart должен рендериться без ошибок 1`] = `
<Fragment>
  <div
    className="src_shared_components_costestimate_v__components_chart_styles_css__legend"
  >
    <ChartLegend />
  </div>
  <div
    className="src_shared_components_costestimate_v__components_chart_styles_css__chart"
    style={
      Object {
        "height": 186,
      }
    }
  >
    <canvas />
  </div>
</Fragment>
`;
